mrmap/registry/tests/acceptance/steps/steps.py

```diff
@@ -20,3 +23,116 @@ def step_impl(context, json_path):
     results = jsonpath(json_body, json_path)
     if results:
         fail('Match found at <{path}> for <{body}>'.format(path=json_path, body=json_body))
+
+
+@then('all {the_class} objects are returned')
+def get_all_resources(context, the_class):
+    if the_class == 'services':
+        total_number_of_features = Service.objects.count()
+    elif the_class == 'layers':
+        total_number_of_features = Layer.objects.count()
+    elif the_class == 'featuretypes':
+        total_number_of_features = FeatureType.objects.count()
+    elif the_class == 'dataset_metadata':
+        total_number_of_features = DatasetMetadata.objects.count()
+    else:
+        total_number_of_features = 0
+
+    response = context.response
+    results = response.json()
+
+    # the response is actually paginated, and if more than 20 results are in total number of features,
+    # then len(results['results']) will be 20
+    if int(results['count']) > 20:
+        assert_that(len(results['results'])).is_equal_to(20)
+        assert_that(int(results['total_pages'])).is_equal_to(math.ceil(int(total_number_of_features) / len(results['results'])))
+    else:
+        assert_that(len(results['results'])).is_equal_to(int(total_number_of_features))
+    assert_that(int(results['count'])).is_equal_to(int(total_number_of_features))
+
+
+# TODO: refactor to be only one step that makes the call and gives the response, instead of giving the url in a
+#  previous step. Needs to be implemented. Currently using the method used by Behave-Rest
+@then('{number_of_resources_returned} result(s) is(are) returned, with {search_criteria} as {filter_keyword} criteria')
+def get_some_resources_by_search(context, number_of_resources_returned, search_criteria, filter_keyword):
+    response = context.response
+    results = response.json()
+
+    # the response is actually paginated, and if more than 20 results are in total number of features,
+    # then len(results['results']) will be 20
+    if int(results['count']) > 20:
```

Comment:
since the response comes paginated every time, I would say yes.
The only way to get the non paginated response (at least as far as I know) is to:

- Remove the paginated response from the viewset
- Create a separate viewset/endpoint for the test

Therefore, I always need to take the pagination into account

Reply:
I understand that we need to take the pagination into account. How about avoiding the duplicated code by introducing a helper method `assert_result_size` that does the lifting then?